### PR TITLE
Rename no_gradient_checkpoint -> gradient_checkpointing

### PR DESCRIPTION
**Pull request by erictang000**
Addresses https://github.com/anyscale/llm-forge/issues/495 and makes changes in template configs

### CODE REVIEW

**Review by kouroshHakha — COMMENTED** [review]
This should wait until there is a new release of llmforge.

**Review by SumanthRH — APPROVED** [review]
@erictang000 can you run a quick test for one of the configs here with the latest llmforge image? Should be good to merge after that

**Comment by SumanthRH** [comment]
Thanks! IMO that config was supposed to be updated after 0.5.6 release!

**Pull request merged**
